cpp/include/cudf/functions.h

```diff
@@ -362,6 +362,16 @@ gdf_error gdf_extract_datetime_month(gdf_column *input, gdf_column *output);
  */
 gdf_error gdf_extract_datetime_day(gdf_column *input, gdf_column *output);
 
+/**
+ * @brief  Extracts day from any date time type and places results into a preallocated GDF_INT16 column
+ *
+ * @param[in] gdf_column of the input
+ * @param[out] output gdf_column. The output memory needs to be preallocated
+ *
+ * @returns GDF_SUCCESS upon successful compute, otherwise returns appropriate error code
+ */
+gdf_error gdf_extract_datetime_weekday(gdf_column *input, gdf_column *output);
```

Comment:
We no longer put APIs in `functions.h`. See https://github.com/rapidsai/cudf/blob/branch-0.10/cpp/docs/TRANSITIONGUIDE.md for the guide on API design. 

Suggested new API:
```
namespace cudf{
gdf_column extract_datetime_weekday(gdf_column const& input);
}
```

Reply:
fixed, let me know if this is what you meant.